File: fincapp-backend/src/FincApp.Core/Animal.cs
using System;
using System.Collections.Generic;

namespace FincApp.Core;

public class Animal
{
    public Guid Id { get; set; }
    public Guid FarmId { get; set; }
    public ProductionType Type { get; set; }
    public string IdentificationTag { get; set; } = string.Empty;
    public DateTime? BirthDate { get; set; }
    public string Status { get; set; } = "healthy";
    public DateTime CreatedAt { get; set; }

    public Farm? Farm { get; set; }
    public ICollection<WeightLog> WeightLogs { get; set; } = new List<WeightLog>();
    public ICollection<HealthRecord> HealthRecords { get; set; } = new List<HealthRecord>();
}

File: fincapp-frontend/mobile-app/app/src/main/java/com/irwi/fincapp/models/Animal.java
package com.irwi.fincapp.models;

public class Animal {
    public long id;
    public String cloudId;
    public String farmCloudId;
    public String type;
    public String identificationTag;
    public String birthDate;
    public String status;
    public String createdAt;
    public String syncStatus;

    public Animal(long id, String cloudId, String farmCloudId, String type, String identificationTag,
                  String birthDate, String status, String createdAt, String syncStatus) {
        this.id = id;
        this.cloudId = cloudId;
        this.farmCloudId = farmCloudId;
        this.type = type;
        this.identificationTag = identificationTag;
        this.birthDate = birthDate;
        this.status = status;
        this.createdAt = createdAt;
        this.syncStatus = syncStatus;
    }
}
